bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServiceTest.java

```diff
@@ -642,7 +642,7 @@ public void testEvents() {
         AtomicReference<List<Event>> events = new AtomicReference<>(new ArrayList<>());
         ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
         waitForAssert(() -> {
-            verify(mockPublisher, atLeast(SEQUENCE.length)).post(eventCaptor.capture());
+            verify(mockPublisher, atLeast(SEQUENCE.length + 1)).post(eventCaptor.capture());
```

Comment:
please leave a comment in the code why there should be 5 invocations. Maybe also on line 639 because its not obvious that there is an event generated internally.

Reply:
Comments added